### PR TITLE
Add and use symbolt constructors

### PR DESCRIPTION
**Pull request by tautschnig**
To the extent possible, apply resource-acquisition-is-initialisation.
The constructors ensure that at least the most essential fields (name,
type, mode) are set.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by tautschnig** [comment]
To reviewers (multiple required): please review commit-by-commit, possibly choosing only those parts of the code that you actually care about (the parts can be inferred from the commit messages).

File: jbmc/src/java_bytecode/java_bytecode_convert_class.cpp [hunk 1]
```diff
@@ -446,14 +446,10 @@ void java_bytecode_convert_classt::convert(
   }(id2string(c.name));
 
   // produce class symbol
-  symbolt new_symbol;
+  class_type.set_name(qualified_classname);
+  type_symbolt new_symbol{qualified_classname, class_type, ID_java};
```
Comment (chrisr-diffblue) [inline jbmc/src/java_bytecode/java_bytecode_convert_class.cpp:453]: 👍

**Review by chrisr-diffblue — APPROVED** [review]
Whilst I'm not usually a massive fan of wide changes for the sake of them, I'm very happy to see this change - especially the dropping of lots of `...is_type=false` :-) 

Might be worth a quick sanity check with @martin-cs that this doesn't clash with anything urgent he may be working on - especially around gnat2goto and related projects.

**Review by feliperodri — APPROVED** [review]
Approving changes to code contracts. LGTM.

**Review by chris-ryder — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Comment by tautschnig** [comment]
~~Requires #7160 for CI to pass.~~ (merged, rebased)

**Pull request merged**